yq.go

```diff
@@ -241,6 +242,41 @@ Note that if you set both flags only overwrite will take effect.
 	return cmdMerge
 }
 
+func createLintCmd() *cobra.Command {
+	var cmdLint = &cobra.Command{
+		Use:     "lint [yaml_file]",
+		Aliases: []string{"l"},
+		Short:   "yq l sample.yaml",
+		Example: `
+yq lint sample.yaml
+yq l sample.yaml
+      `,
+		Long: "Does a basic lint on the file to see if it can be parsed",
+		RunE: lint,
+	}
+	return cmdLint
+}
+
+func lint(cmd *cobra.Command, args []string) error {
+	var path = ""
+	if len(args) < 1 {
+		return errors.New("Must provide filename")
+	} else if len(args) >= 1 {
+		path = args[0]
+	}
+	obj := make(map[string]interface{})
```

Comment:
Yaml files are not necessarily map[string]interface{}, you can also have arrays:
```
- 1
- 2
```

single values

```
cat
```

multi-doc
```
a: thing
---
- 1
- 2
```

Reply:
You may be better off reusing/refactoring the readProperty function, currently if you don't give it a path it will just spit out the original yaml. Maybe you could extract out the reading functionality it so the lint command runs the same underlying extracted function, but does not print the contents or give a path, and instead just prints out parsing errors if there are any and 'ok' otherwise...

Reply:
That makes sense.  Let me see what I can do.  Thanks!